WordPress/src/test/java/org/wordpress/android/ui/qrcodeauth/QRCodeAuthViewModelTest.kt

```diff
@@ -1,30 +1,553 @@
+import org.wordpress.android.fluxc.network.rest.wpcom.qrcodeauth.QRCodeAuthErrorType.INVALID_RESPONSE
+import org.wordpress.android.fluxc.network.rest.wpcom.qrcodeauth.QRCodeAuthErrorType.NOT_AUTHORIZED
+import org.wordpress.android.fluxc.store.qrcodeauth.QRCodeAuthStore
+import org.wordpress.android.fluxc.store.qrcodeauth.QRCodeAuthStore.QRCodeAuthAuthenticateResult
+import org.wordpress.android.fluxc.store.qrcodeauth.QRCodeAuthStore.QRCodeAuthResult
+import org.wordpress.android.fluxc.store.qrcodeauth.QRCodeAuthStore.QRCodeAuthValidateResult
+import org.wordpress.android.ui.posts.BasicDialogViewModel.DialogInteraction
+import org.wordpress.android.ui.qrcodeauth.QRCodeAuthUiState.Content.Done
+import org.wordpress.android.ui.qrcodeauth.QRCodeAuthUiState.Content.Validated
+import org.wordpress.android.ui.qrcodeauth.QRCodeAuthUiState.Error.InvalidData
+import org.wordpress.android.ui.qrcodeauth.QRCodeAuthUiState.Error.NoInternet
+import org.wordpress.android.ui.qrcodeauth.QRCodeAuthUiStateType.AUTHENTICATING
+import org.wordpress.android.ui.qrcodeauth.QRCodeAuthUiStateType.AUTH_FAILED
+import org.wordpress.android.ui.qrcodeauth.QRCodeAuthUiStateType.DONE
+import org.wordpress.android.ui.qrcodeauth.QRCodeAuthUiStateType.EXPIRED
+import org.wordpress.android.ui.qrcodeauth.QRCodeAuthUiStateType.INVALID_DATA
+import org.wordpress.android.ui.qrcodeauth.QRCodeAuthUiStateType.NO_INTERNET
+import org.wordpress.android.ui.qrcodeauth.QRCodeAuthUiStateType.VALIDATED
+import org.wordpress.android.ui.qrcodeauth.QRCodeAuthViewModel.Companion.BROWSER_KEY
+import org.wordpress.android.ui.qrcodeauth.QRCodeAuthViewModel.Companion.DATA_KEY
+import org.wordpress.android.ui.qrcodeauth.QRCodeAuthViewModel.Companion.LAST_STATE_KEY
+import org.wordpress.android.ui.qrcodeauth.QRCodeAuthViewModel.Companion.LOCATION_KEY
+import org.wordpress.android.ui.qrcodeauth.QRCodeAuthViewModel.Companion.TOKEN_KEY
 import org.wordpress.android.util.NetworkUtilsWrapper
 
-@RunWith(MockitoJUnitRunner::class)
-class QRCodeAuthViewModelTest {
-    @Rule
-    @JvmField val rule = InstantTaskExecutorRule()
+@InternalCoroutinesApi
+@ExperimentalCoroutinesApi
+@Suppress("LargeClass")
+class QRCodeAuthViewModelTest : BaseUnitTest() {
+    private lateinit var viewModel: QRCodeAuthViewModel
+    @Mock lateinit var store: QRCodeAuthStore
     @Mock lateinit var networkUtilsWrapper: NetworkUtilsWrapper
+    @Mock lateinit var savedInstanceState: Bundle
     @Mock lateinit var validator: QRCodeAuthValidator
     private val uiStateMapper = QRCodeAuthUiStateMapper()
 
-    private lateinit var viewModel: QRCodeAuthViewModel
+    private val data = "data"
+    private val location = "location"
+    private val browser = "browser"
+    private val token = "token"
+    private val scannedValue =
+            "https://apps.wordpress.com/get/?campaign=login-qr-code#qr-code-login?token=scannedtoken&data=scanneddata"
```

Comment:
Can we convert them to `Constants` (usually put above `QRCodeAuthViewModelTest` class)?

Reply:
Sure, in commit [79434b2](https://github.com/wordpress-mobile/WordPress-Android/pull/16713/commits/79434b2b23d8a2b02befe0f04f24d45ea0908906).  I've seen these both ways in tests - constants and private members. How do you decide which one to go with?

Reply:
>  I've seen these both ways in tests 

Yeah, I've noticed this inconsistency as well. I personally prefer making them `Constants` if the value is known at compile-time and guaranteed not to be changed.